### ISSUE TITLE
Do not allow to delete product attributes when they are used in attribute combinations

### ISSUE THREAD

**Issue by dtaalbers** [nopSolutions/nopCommerce#3950]
nopCommerce version: 4.20

Steps to reproduce the problem: 

1) Add two product attributes
 ```
* Color with values: Yellow, Blue
* Ram with values: 1GB, 2GB
```

![Screen Shot 2019-06-13 at 13 26 45](https://user-images.githubusercontent.com/6512556/59429137-66986e80-8ddf-11e9-968d-7c53be58e843.png)

2) Generate all possible combinations

![Screen Shot 2019-06-13 at 13 26 57](https://user-images.githubusercontent.com/6512556/59429163-74e68a80-8ddf-11e9-9537-d55cc153c2fe.png)

3) Remove attribute RAM

![Screen Shot 2019-06-13 at 13 28 03](https://user-images.githubusercontent.com/6512556/59429230-a2333880-8ddf-11e9-8502-5b7d798aa224.png)

4) You are left with duplicate attribute combinations

![Screen Shot 2019-06-13 at 13 27 11](https://user-images.githubusercontent.com/6512556/59429198-9182c280-8ddf-11e9-94b9-a45174d9908c.png)

Not sure if you classify this as a bug. But doesn't seem right to me, so thats why I am reporting it 😉 

Andrei M - Let's display some warning (e.g. "This product attribute is used in some products. It's highly recommended to update appropriate product details pages first") 
with confirmation  when deleting such product attribute. It'll be enough

**Comment by kwerty** [comment]
If you guys are planning to soft-delete attribute values, perhaps you might also consider an "Is Active" property for attribute values?

There are times when we run out of a particular colour and we're not expecting to have that colour in stock for a while. When this happens we'd prefer to be able to hide the value rather than delete it.

**Comment by kwerty** [comment]
Furthermore, there is a bug when deleting attribute combinations.

* Customer adds product to their cart with an attribute combination
* Admin deletes attribute combination

Current behavior: User is still able to checkout, even though the attribute combination no longer exists.

Expected behavior: Error should be displayed on shopping cart page, notifying user that they need to select a different combination of attributes.

**Comment by holydk** [comment]
Closed #3950
